subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java

```diff
@@ -287,6 +289,9 @@ protected ScriptPluginFactory createScriptPluginFactory(ScriptingLanguages scrip
     }
 
     private DefaultScriptPluginFactory defaultScriptPluginFactory() {
+        PluginRequestsTransformer compositeTransformer = new CompositePluginRequestsTransformer(
+            getAll(PluginRequestsTransformer.class)
+        );
```

Comment:
:x: Same thing, this will only apply to Groovy DSL scripts, see my other comment above.

Reply:
I would introduce a new `PluginRequestsTransformer(Pipeline|Aggregator)`  service endpoint to abstract away the composition and then use it on the Kotlin and Groovy `ScriptPluginFactory` implementations.